pkg/discovery-api/discovery_api.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2022 FairwindsOps Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2020 Fairwinds
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+package discoveryapi
+
+import (
+	"context"
+	"encoding/json"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	"github.com/fairwindsops/pluto/v5/pkg/api"
+)
+
+// DiscoveryClient is the
+type DiscoveryClient struct {
+	ClientSet       dynamic.Interface
+	restConfig      *rest.Config
+	DiscoveryClient discovery.DiscoveryInterface
+	Instance        *api.Instance
+}
+
+// NewDiscoveryClient returns a new struct with config portions complete.
+func NewDiscoveryClient(instance *api.Instance) *DiscoveryClient {
+	cl := &DiscoveryClient{
+		Instance: instance,
+	}
+
+	var err error
+	cl.restConfig, err = NewRestClientConfig(rest.InClusterConfig)
+	if err != nil {
+		return nil
```

Comment:
would it be better for this function to actually return the errors and then handle them when calling from the root.go? 

Currently, if connecting to a cluster fails, this exits cleanly with confusing output. Example:

```
▶ kubectl get nodes
The connection to the server 127.0.0.1:59986 was refused - did you specify the right host or port?

✗ ./pluto detect-api-resources
There were no resources found with known deprecated apiVersions.

✗ echo $?
0
```

Reply:
Corrected.